mongo/mongo.go

```diff
@@ -0,0 +1,365 @@
+	txnOptions := options.Transaction().SetWriteConcern(wc)
+
+	session, err := db.db.Client().StartSession()
+	if err != nil {
+		return fmt.Errorf("failed starting session to: %w", err)
+	}
+	defer session.EndSession(context.Background())
+
+	err = mongo.WithSession(context.Background(), session, func(ctx mongo.SessionContext) error {
+		if err = session.StartTransaction(txnOptions); err != nil {
+			return fmt.Errorf("failed to pending transaction: %w", err)
+		}
+
+		err = db.executeTransactions(ctx, tx, session)
+		if err != nil {
+			return err
+		}
+
+		if err = session.CommitTransaction(ctx); err != nil {
+			return fmt.Errorf("failed committing transaction to: %w", err)
+		}
+		return nil
+	})
+
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// CreateTable creates a collection or an embedded collection if it does not exists.
+func (db *DB) createTable(ctx context.Context, collection []byte) error {
+	if err := db.db.CreateCollection(ctx, string(collection)); err != nil {
+		return fmt.Errorf("failed creating collection %s to: %w", collection, err)
+	}
+
+	// create an index on the Key field
+	index := mongo.IndexModel{
+		Keys:    createFilter("key", 1),
+		Options: options.Index().SetUnique(true),
+	}
+
+	_, err := db.db.Collection(string(collection)).Indexes().CreateOne(ctx, index)
+	if err != nil {
+		return fmt.Errorf("failed creating collection %s to: %w", collection, err)
+	}
+
+	return nil
+}
+
+// DeleteTable deletes a root or embedded collection. Returns an error if the
+// collection cannot be found or if the key represents a non-collection value.
+func (db *DB) deleteTable(ctx context.Context, collection []byte) error {
+	if !collectionExists(ctx, db.db, string(collection)) {
+		return fmt.Errorf("failed deleting collection %s to: %w ", collection, database.ErrNotFound)
+	}
+
+	if err := db.db.Collection(string(collection)).Drop(ctx); err != nil {
+		return fmt.Errorf("failed dropping collection %s to: %w", collection, err)
+	}
+	return nil
+}
+
+func (db *DB) get(ctx context.Context, collection, key []byte) (ret []byte, err error) {
+	filter := createFilter("key", key)
+	res := db.db.Collection(string(collection)).FindOne(ctx, filter)
+
+	if err := res.Err(); err != nil {
+		return nil, fmt.Errorf("failed finding %s/%s to: %w", collection, key, database.ErrNotFound)
+	}
+
+	result := tuple{}
+	if err := res.Decode(&result); err != nil {
+		return nil, fmt.Errorf("failed decoding value for %s/%s to: %w", collection, key, err)
+	}
+
+	return result.Value, nil
+}
+
+func (db *DB) set(ctx context.Context, collection, key, value []byte) error {
+	filter := createFilter("key", key)
+	update := createUpdate(value)
+	opts := options.Update().SetUpsert(true)
+
+	_, err := db.db.Collection(string(collection)).UpdateOne(ctx, filter, update, opts)
+	if err != nil {
+		return fmt.Errorf("failed setting value %s/%s to: %w", collection, key, err)
+	}
+	return nil
+}
+
+// List returns the full list of entries in a collection.
```

Comment:
There's a few places where the comment is outdated with the current implementation in terms of exported vs. non-exported.